aait/backend/group-2B/WebApi/Controllers/PostsController.cs

```diff
@@ -11,56 +13,64 @@ namespace WebApi.Controllers
     public class PostsController : ControllerBase
     {
         private readonly IMediator _mediator;
+        private readonly IUserService _userService;
 
-        public PostsController(IMediator mediator)
+        public PostsController(IMediator mediator, IUserService userService)
         {
             _mediator = mediator;
+            _userService = userService;
         }
 
         [HttpGet]
         public async Task<IActionResult> GetAllAsync()
         {
             var getAllRequest = new GetAllPostsRequest();
-            var posts = await _mediator.Send(getAllRequest);
-            return Ok(posts);
+            var response = await _mediator.Send(getAllRequest);
+            return Ok(response.Value);
         }
 
         [HttpGet("{id}")]
         public async Task<IActionResult> Get(int id)
         {
             var getPostByIdRequest = new GetPostByIdRequest { Id = id };
-            var post = await _mediator.Send(getPostByIdRequest);
-            return Ok(post);
+            var response = await _mediator.Send(getPostByIdRequest);
+            return Ok(response.Value);
         }
 
         [HttpPost]
+        [Authorize]
         public async Task<IActionResult> CreatePostAsync([FromBody] CreatePostDto createPostDto)
         {
+            int userId = _userService.GetUserId();
+            createPostDto.UserId  = userId;
             var createCommand = new CreatePostCommand { CreatePostDto = createPostDto };
             var response = await _mediator.Send(createCommand);
             if (response.IsSuccess)
             {
-                return CreatedAtAction(nameof(Get), new { id = response.Value }, response);
+                return CreatedAtAction(nameof(Get), new { id = response.Value }, response.Value);
             }
             else
             {
-                return BadRequest(response.Message);
+                return BadRequest(response.Error);
             }
         }
 
         [HttpPatch]
```

Comment:
The best practices for RESTful APIs' delete and update verbs is to include the id of the resource to be updated/deleted in the endpoint. Like `PATCH http://endpoint.api/user/1` or `DELETE https://endpoint.api/users/1`